src/vs/editor/browser/core/editorState.ts

```diff
@@ -164,17 +164,15 @@ export class StableEditorScrollState {
 		}
 	}
 
-	public restoreRelativeVerticalPositionOfCursor(editor: ICodeEditor, shouldPushUndoStop: boolean = false): void {
+	public restoreRelativeVerticalPositionOfCursor(editor: ICodeEditor): void {
 		const currentCursorPosition = editor.getPosition();
 
 		if (!this._cursorPosition || !currentCursorPosition) {
 			return;
 		}
 
 		const offset = editor.getTopForLineNumber(currentCursorPosition.lineNumber) - editor.getTopForLineNumber(this._cursorPosition.lineNumber);
-		if (shouldPushUndoStop) {
-			editor.pushUndoStop();
-		}
 		editor.setScrollTop(editor.getScrollTop() + offset);
+		editor.pushUndoStop();
```

Comment:
As I just said above, this call does not belong here. Please remove it.

Also note the caller side where an undo stop is pushed only conditionally.